pkg/operator/controllers/guardrails/gktemplates/aro-deny-delete.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: arodenydelete
+  annotations:
+    description: >-
+      Prevents unexpected deletion of a specific critical resource
+spec:
+  crd:
+    spec:
+      names:
+        kind: ARODenyDelete
+      validation:
+        openAPIV3Schema:
+          type: object
+          properties:
+            name:
+              type: string
+              description: >-
+                The specific resource name that is not allowed to be deleted.
+  targets:
+  - target: admission.k8s.gatekeeper.sh
+    rego: |
+      package arodenydelete
+
+      violation[{"msg": msg}] {
+        input.review.operation == "DELETE"
+        deny_resource_name := input.parameters.name
+        deny_resource_name == input.review.object.metadata.name
+        msg := "Deleting resource - " + name + " is not allowed"
```

Comment:
this does not work:
```
template:6: rego_type_error: plus: invalid argument(s)
	have: (number, string, ???)
	want: (x: number, y: number, z: number)
template:6: rego_type_error: plus: invalid argument(s)
	have: (string, any, ???)
	want: (x: number, y: number, z: number) 
```
and has to be something like:
`        msg := sprintf("Deleting resource - %v is not allowed", [deny_resource_name])
`